contractcourt/breacharbiter.go

```diff
@@ -1441,7 +1441,7 @@ func (b *BreachArbiter) sweepSpendableOutputsTxn(txWeight int64,
 
 	// Create a sighash cache to improve the performance of hashing and
 	// signing SigHashAll inputs.
-	hashCache := txscript.NewTxSigHashes(txn)
+	hashCache := txscript.NewTxSigHashesV0Only(txn)
```

Comment:
@Roasbeef: This whole commit makes me a bit uneasy. And we'll probably have to add way more logic. Can't we somehow simplify this in `btcd` itself? Since this seems to only be needed for SigHashAnyoneCanPay and we never do that in `lnd` anyway?

Reply:
This `NewTxSigHashesV0Only` function doesn't exist rn right? What do you mean it makes you uneasy? If we specify the `CannedPrevOutFetcher` (or w/e it's called) here, we make sure we're always doing the right thing. In each instance, we also already know what the prev out is.

Reply:
I moved the `NewTxSigHashesV0Only` from my `btcd` branch into `lnd` and updated most of the non-test code to not actually use it (and set up a proper prev output fetcher instead).
Turned out we do have all the info about all the spent outputs in pretty much all cases. which makes me feel way better about this. The `NewTxSigHashesV0Only ` is only used in unit test code or cases where we _know_ we currently only have v0 inputs (e.g. a channel funding output).

Reply:
Oops, yeah, I forgot to update that commit. Fixed.